Framework/Core/src/Kernels.cxx

```diff
@@ -53,24 +53,27 @@ Status doGrouping(std::shared_ptr<ChunkedArray> chunkedArray, Datum* outputRange
 {
   TableBuilder builder;
   auto writer = builder.persist<uint64_t, uint64_t, uint64_t>({"start", "count", "index"});
-  auto chunk = std::static_pointer_cast<ARRAY>(chunkedArray->chunk(0));
-  if (chunk->length() == 0) {
+  auto zeroChunk = std::static_pointer_cast<ARRAY>(chunkedArray->chunk(0));
+  if (zeroChunk->length() == 0) {
     *outputRanges = std::move(builder.finalize());
     return arrow::Status::OK();
   }
-  T currentIndex = chunk->raw_values()[0];
-  T currentCount = 0;
-  T currentOffset = 0;
-  for (size_t ci = 0; ci < chunkedArray->num_chunks(); ++ci) {
+  uint64_t currentIndex = 0;
+  uint64_t currentCount = 0;
+  uint64_t currentOffset = 0;
```

Comment:
Why do you need this? In principle we could use smaller integers to keep track of ranges.

Reply:
@ktf this is definitely not the best option, however these numbers are filled into arrow table that is uint64 anyway. Which, in turn, happens because we need to extract the raw arrays from it and need to have certain type. I will remake this completely - within one helper class we can avoid passing arguments and deducing types - but first I want to make sure the grouping is viable with this approach.